### PR TITLE
feat: [ANDROAPP-3206] New event icons

### PR DESCRIPTION
**Pull request by Balcan**
## Description
[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3206)

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt [hunk 1]
```diff
@@ -51,10 +53,98 @@ class ProgramEventMapper @Inject constructor(val d2: D2) {
...
             hasExpired || !inOrgUnitRange,
             attrOptCombo,
-            event.geometry()
+            event.geometry(),
+            isEventEditable(event.uid())
         )
     }
 
+    fun isEventEditable(eventUid: String): Boolean {
+        val event =
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt:62]: can you extract this to a class? so is ready to move for sdk.. :dancers:

Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt:62]: I was thinking about it. I'll do it yes :)

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureRepositoryImpl.java [hunk 2]
```diff
@@ -70,14 +68,15 @@
...
-    private String lastUpdatedUid;
     private RuleEvent.Builder eventBuilder;
     private List<FieldViewModel> sectionFields;
+    private final DhisEventUtils dhisEventUtils;
 
     public EventCaptureRepositoryImpl(FieldViewModelFactory fieldFactory, FormRepository formRepository, String eventUid, D2 d2) {
         this.eventUid = eventUid;
         this.formRepository = formRepository;
         this.d2 = d2;
+        this.dhisEventUtils = new DhisEventUtils(d2);
```
[the remark this answers is not in the document]
Comment (JaimeToca) [reply app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureRepositoryImpl.java:79]: yes, we could not mock it for the future

File: app/src/main/java/org/dhis2/data/event/DhisEventUtils.kt [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnit
+
+class DhisEventUtils @Inject constructor(val d2: D2) {
+
+    fun isEventEditable(eventUid: String): Boolean {
+        val event =
+            d2.eventModule().events().uid(eventUid).blockingGet()
+        val program =
+            d2.programModule().programs().uid(event.program()).blockingGet()
+        val stage =
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/event/DhisEventUtils.kt:19]: I would go a bit further and have isEnrollmentOpen, inOrgUnitRange, getCatComboAccess etc.. in different classes.
Then you could easily add unit test :dancing_women: XD

Comment (Balcan) [reply app/src/main/java/org/dhis2/data/event/DhisEventUtils.kt]: I have all different classes. The tests are too much to get this ready now. So i'll do it next week.

**Pull request merged**